src/test/java/com/puppycrawl/tools/checkstyle/grammar/GeneratedJavaTokenTypesTest.java

```diff
@@ -237,24 +237,27 @@ public void testTokenNumbering() {
         Assert.assertEquals(message, 181, GeneratedJavaTokenTypes.LAMBDA);
         Assert.assertEquals(message, 182, GeneratedJavaTokenTypes.BLOCK_COMMENT_END);
         Assert.assertEquals(message, 183, GeneratedJavaTokenTypes.COMMENT_CONTENT);
-        Assert.assertEquals(message, 184, GeneratedJavaTokenTypes.SINGLE_LINE_COMMENT_CONTENT);
-        Assert.assertEquals(message, 185, GeneratedJavaTokenTypes.BLOCK_COMMENT_CONTENT);
```

Comment:
Read javadoc of this test. These numbers **CAN'T** change. They must remain as they are.

Reply:
I read it. The thing is that generated tokens are ordered by visibility and new token is placed in the middle of the list.

I can try to make it protected to put it to the end of the list, but it seems odd.

Reply:
The javadoc explains why it must be like this. There is no alternative unless something like  https://github.com/checkstyle/checkstyle/issues/5880 can be implemented. Until then, pre-existing numbers can't change.

Reply:
@rnveach I tried to 
1. add protected modifier
2. remove it from tokens list in the beginning of the file

In both cases generated number is 184.
The other option is to include multiline strings in STRING_LITERAL token, but I dont like it.

Reply:
@romani ping
I thought adding new tokens to the end of the list of the G file was all that was needed?

Reply:
Update: parsing wont work with protected modifier, it should not be protected. But regardless modifier, generated number is 184 on my PC.

Reply:
I think I see issue now.

https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g#L111
Along with adding `MULTILINE_STRING_LITERAL` to the end, other tokens whose number are changing need to be added to the end first. Since they aren't listed there, ANTLR is allowing their numbers to come after `MULTILINE_STRING_LITERAL` instead of before.

So you will need to add like:
````
SINGLE_LINE_COMMENT_CONTENT;
BLOCK_COMMENT_CONTENT;
....
MULTILINE_STRING_LITERAL
````

Reply:
This works, thank you.
I think it should be mentioned somewhere, because this is not obvious and clear.
I corrected order, should be ok now.

Reply:
Feel free to update the Javadoc for the test. There is no other area that specifically deals with tokens and their numbers.

Reply:
done, please take a look

Reply:
Corrected javadoc and CI is green